test/mocks/http/conn_pool.h

```diff
@@ -0,0 +1,43 @@
+#include <memory>
+
+#include "envoy/http/conn_pool.h"
+
+#include "gmock/gmock.h"
+
+namespace Envoy {
+
+namespace Upstream {
+class MockHostDescription;
```

Comment:
is there any specific reason to do forward declaration? if no just include.

Reply:
Mostly to avoid including headers in headers. I hear it's bad practice nowadays for a few reasons. I'll clean it up.

Reply:
FYI: https://google.github.io/styleguide/cppguide.html#Forward_Declarations
if no strong reason (we usually do it for avoid cyclic dependency of bazel targets), just include.

Reply:
Yeah, I read that after making this change. Meant to clean it up, but I clearly forgot to. Their discussion as to why it's bad convinced me that I shouldn't be doing it.

I can't wait for C++20 modules. I *think* they'll clean this stuff up. :)